test/fake-intake/fakeintake/fakeintake.go

```diff
@@ -66,71 +68,89 @@ type postPayloadResponse struct {
 	Errors []string `json:"errors"`
 }
 
-func (fi *FakeIntake) postPayload(w http.ResponseWriter, req *http.Request) {
-	fmt.Println("Handling PostPayload request")
+func (fi *FakeIntake) handleDatadogRequest(w http.ResponseWriter, req *http.Request) {
 	if req == nil {
-		writePostPayloadResponse(w, []string{"invalid request, nil request"})
+		response := buildPostResponse(errors.New("invalid request, nil request"))
+		writeHttpResponse(w, response)
 		return
 	}
 
+	log.Printf("Handling Datadog %s request to %s", req.Method, req.URL.Path)
+
+	if req.Method == http.MethodGet {
```

Comment:
You can avoid checks like that by using a more advanced `mux`, for instance:
https://github.com/DataDog/datadog-agent/blob/main/cmd/cluster-agent/api/v1/clusterchecks.go#L28-L33

(same with POST)

Reply:
Didn't know we were using gorilla mux, I sticked to the standard libs without looking into code for gorilla usages. Can I move to gorilla in a follow up PR ?